tpm2/src/seal/main.c

```diff
@@ -253,19 +260,20 @@ int main(int argc, char **argv)
                       authString, pcrsString, ownerAuthPasswd, cipherString))
   {
     kmyth_log(LOG_ERR, "kmyth-seal error ... exiting");
-    free(outPath);
     if (authString != NULL)
     {
       kmyth_clear(authString, strlen(authString));
     }
     kmyth_clear(ownerAuthPasswd, strlen(ownerAuthPasswd));
+    free(outPath);
     return 1;
   }
 
   // Clean-up any remaining resources
   //   Note: authString and ownerAuthPasswd cleared after use in
```

Comment:
This is not a comment on your proposed changes, but in reviewing this, I realized that our authString and ownerAuthPasswd freeing and clearing approach is different for seal and unseal. In seal, we clear authString and ownerAuthPasswd as soon as they are no longer needed (within tpm2_kmyth_seal()). I think it is good to clear them earlier rather than later, but it is less transparent to the main process (e.g., we are not freeing/clearing in the same code where the variables are declared/created. I can see arguments both ways, but it seems to me that our approach should be consistent.

Reply:
I am happy to update this either way, when a decision is reached. Alternatively, could we clear authString and ownerAuthPasswd in both the library functions and the mains?